include/ck_tile/core/container/thread_buffer.hpp

```diff
@@ -59,6 +59,14 @@ struct thread_buffer {
     template <index_t I> CK_TILE_HOST_DEVICE constexpr auto& at(number<I>)              { return get(I); }
     template <index_t I> CK_TILE_HOST_DEVICE constexpr const auto& at(number<I>) const  { return get(I); }
 
+    CK_TILE_HOST_DEVICE constexpr void set_value(index_t i, const value_type& value) { 
```

Comment:
we have `set_as()` API in below, do we need a new one here? I feel you can just reuse that in below

Reply:
Fixed